docs/manual/src/swift/xcode.md

```diff
@@ -1,8 +1,52 @@
 # Integrating with XCode
 
-It is possible to generate Swift bindings at compile time for XCode projects.
-In your project configuration, add a "Run Script Phase" that applies to `*.udl` files:
+It is possible to generate Swift bindings at compile time for XCode projects
+and incorporate them alongside hand-written Swift code to form a larger module.
+Broadly, you will need to:
+
+1. Add a build phase to compile the Rust crate into a static lib and link it
+   into your framework.
+2. Add a build phase to run `uniffi-bindgen ` and generate the Swift bindings.
+3. Include the generated bridging header into your overall bridging header.
+
+There is also an example app in the UniFFI project repo that may be helpful.
+
+## Compiling the Rust crate.
+
+Sorry, configuring XCode to compile the Rust crate into a staticlib
+is beyond the scope of this document. However you do so, make sure you
+include the resulting `libexample.a` file in the "Link Binary with Libraries"
+build phase for your framework.
+
+## Generating the bindings
+
+In the "Build Rules" section of your config, add a rule to process `.udl` files
+using `uniffi-bindgen`. We recommend having it generate the output files
+somewhere in your source tree, rather than in XCode's default `$DERIVED_FILE_DIR`;
+this both helps with debugging the build output, and makes it easier to configure
+how the generated files are used.
+
+* Add a build rule processing files with names matching `*.udl`.
+    * Use something like the following as the custom script:
+        * `$HOME/.cargo/bin/uniffi-bindgen generate $INPUT_FILE_PATH --language swift --out-dir $INPUT_FILE_DIR/Generated`
+    * Add both the `.swift` file and the generated bridging header as output files:
+        * `$(INPUT_FILE_DIR)/Generated/$(INPUT_FILE_BASE).swift`
+        * `$(INPUT_FILE_DIR)/Generated/uniffi_$(INPUT_FILE_BASE)-Bridging-Header.h`
+* Add your `.udl` file to the "Compile Sources" build phase for your framework,
+  so that XCode will process it using the new build rule and will include the resulting
+  outputs in the build.
+
+## Including the bridging header
+
+In the overall bridging header for your module, include the bridging header
+generated by UniFFI in the previous step:
 
-```bash
-$HOME/.cargo/bin/uniffi-bindgen generate $INPUT_FILE_PATH --language swift --out-dir $DERIVED_FILE_DIR\n
 ```
+#include "uniffi_example-Bridging-Header.h"
+```
+
+For this to work without complaint from XCode, you also need to add the
+generated header file as a **Public** header in the "Headers" build phase
+of your project (which is why it's useful to generate this file somewhere in
+your source tree, rather than in temporary build directory).
+
```

Comment:
We should probably mention not explicitly compiling the generated swift code here as well.